src/libstore/derivations.hh

```diff
@@ -44,13 +44,24 @@ struct DerivationOutputCAFloating
  */
 struct DerivationOutputDeferred {};
 
+/* Impure output which is moved to a content-addressed location (like
+   CAFloating) but isn't registered as a realization.
+ */
+struct DerivationOutputImpure
+{
+    /* information used for expected hash computation */
+    FileIngestionMethod method;
+    HashType hashType;
+};
+
```

Comment:
I don't think we need this? I didn't think we did, and I went through and indeed it seems this always handled the same way as a CA output.